build_tools/github/trigger_hosting.sh

```diff
@@ -31,10 +31,33 @@ else
      BRANCH=$HEAD_BRANCH
 fi
 
-curl --request POST \
+# Circle CI REST API return HTTP response with 202 status code even when the POST requests fail.
+# Hence we add some handling so that errors are reported on GitHub.
+# For details see: https://circleci.com/docs/api/v2/index.html#operation/triggerPipeline
+CIRCLE_CI_RESPONSE=$(curl --request POST \
      --url https://circleci.com/api/v2/project/gh/$GITHUB_REPOSITORY/pipeline \
      --header "Circle-Token: $CIRCLE_CI_TOKEN" \
      --header "content-type: application/json" \
      --header "x-attribution-actor-id: github_actions" \
      --header "x-attribution-login: github_actions" \
-     --data \{\"branch\":\"$BRANCH\",\"parameters\":\{\"GITHUB_RUN_URL\":\"$GITHUB_RUN_URL\"\}\}
+     --data \{\"branch\":\"$BRANCH\",\"parameters\":\{\"GITHUB_RUN_URL\":\"$GITHUB_RUN_URL\"\}\})
+
+echo $CIRCLE_CI_RESPONSE
```

Comment:
I tested this workflow here: https://github.com/thomasjpfan/scikit-learn/actions/runs/3952957085/jobs/6768670140 and with `set -x`, I do not think we need to echo the response. It is already printed.

```suggestion
```